include/tuner/nccl_ofi_tuner_common.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
+
+#ifndef NCCL_OFI_TUNER_COMMON_H_
+#define NCCL_OFI_TUNER_COMMON_H_
+
+#include "config.h"
+
+#include <linux/limits.h>
+#include <nccl/tuner.h>
+
+typedef struct nccl_ofi_tuner_context nccl_ofi_tuner_context_t;
+
+/* region base vs. model base */
+enum nccl_ofi_tuner_type {
+	NCCL_OFI_TUNER_TYPE_REGION = 0,
+	NCCL_OFI_TUNER_TYPE_MODEL
+};
+
+/* platform type for tuner respective */
+enum nccl_ofi_tuner_platform {
+	NCCL_OFI_TUNER_P5_P5E = 0,
+	NCCL_OFI_TUNER_P5EN,
+	NCCL_OFI_TUNER_UNKNOWN,
+	NCCL_OFI_TUNER_PLATFORM_MAX = NCCL_OFI_TUNER_UNKNOWN
+};
+
+struct nccl_ofi_tuner_context {
+	enum nccl_ofi_tuner_type type;
+	/* pointer to tuner type ("Region" or "Model") specific context data */
+	void *type_ctx;
```

Comment:
ick.  I'd much prefer either (1) an inheritance based design like we do with the transports or (2) a union.  (1) is a lot nicer for when we c++-ify all this.

Reply:
ack. (1) would be better looking forward c++ direction. I'll merge this PR as it is for now.